### PR TITLE
23.2 Set Objects

### PR DESCRIPTION
**Pull request by rwaldron**

### CODE REVIEW

**Comment by anba** [comment]
set-iterator-close-after-add-failure.js is a duplicate of set-close-iterator-after-failure.js

File: test/built-ins/Set/prototype/Symbol.iterator/toStringTag-property-descriptor.js [hunk 1]
```diff
@@ -11,7 +11,11 @@
 
 var SetIteratorProto = Object.getPrototypeOf(new Set()[Symbol.iterator]());
 
-assert.sameValue('Set Iterator', SetIteratorProto[Symbol.toStringTag]);
+assert.sameValue(
+ 'Set Iterator',
```
Comment (arv) [inline test/built-ins/Set/prototype/Symbol.iterator/toStringTag-property-descriptor.js]: strange indentation

File: test/built-ins/Set/prototype/add/add.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+es6id: 23.2.3.1
+description: >
+    Set.prototype.add ( value )
+
+    17 ECMAScript Standard Built-in Objects
+
+includes: [propertyHelper.js]
+---*/
+
+assert.sameValue(
```
Comment (arv) [inline test/built-ins/Set/prototype/add/add.js]: Maybe we should have helper for validating standard methods.

We should also validate the `name` and `length` properties and that they do not have a `prototype` property.

```js
verifyBuiltinMethod(Set.prototype, 'add', 1)
```

File: test/built-ins/Set/set-iterator-close-after-add-failure.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+---*/
+
+var count = 0;
+var iterable = {};
+iterable[Symbol.iterator] = function() {
+  return {
+    next: function() {
+      return { value: null, done: false };
+    },
+    'return': function() {
```
Comment (arv) [inline test/built-ins/Set/set-iterator-close-after-add-failure.js]: this is Sparta^d^d^d^d^d^dES6 and we do not have to limit ourselves to ES3 silliness.

**Comment by arv** [comment]
It is too bad we cannot use an control flow structures. Having to repeat all the tests for all 6 primitive types gets verbose.

File: test/built-ins/Set/prototype/Symbol.iterator/toStringTag-property-descriptor.js (hunk 1 above)
Comment (rwaldron) [inline test/built-ins/Set/prototype/Symbol.iterator/toStringTag-property-descriptor.js]: Fixed. I can't tell if this was caused by recast or not :\

File: test/built-ins/Set/prototype/add/add.js (hunk 2 above)
Comment (rwaldron) [inline test/built-ins/Set/prototype/add/add.js]: Sure, but in a different issue

**Comment by rwaldron** [comment]
Is this blocked?

**Comment by bterlson** [comment]
In the sense that I need to make sense of the results yet. I will hopefully get to this later today!

**Comment by rwaldron** [comment]
Ok, that's reasonable

**Comment by bterlson** [comment]
LGTM, squash away!

**Pull request merged**